packages/gatsby-source-drupal/src/normalize.js

```diff
@@ -12,11 +13,132 @@ const getHref = link => {
 
 exports.getHref = getHref
 
-const nodeFromData = (datum, createNodeId, entityReferenceRevisions = []) => {
+const imageCDNState = {
+  foundPlaceholderStyle: false,
+  hasLoggedNoPlaceholderStyle: false,
+}
+
+exports.imageCDNState = imageCDNState
+
+/**
+ * This FN takes in node data and returns Gatsby Image CDN fields that should be added to that node. If the input node isn't an image an empty object is returned.
+ */
+const getGatsbyImageCdnFields = async ({
+  node,
+  type,
+  pluginOptions,
+  fileNodesExtendedData,
+  reporter,
+}) => {
+  if (!pluginOptions?.skipFileDownloads) {
+    return {}
+  }
+
+  const isFile = isFileNode({
+    internal: {
+      type,
+    },
+  })
+
+  if (!isFile) {
+    return {}
+  }
+
+  const mimeType = node.attributes.filemime
+
+  if (!mimeType) {
+    return {}
+  }
+
+  const url = getFileUrl(node.attributes, pluginOptions.baseUrl)?.href
+
+  if (!url) {
+    return {}
+  }
+
+  if (!mimeType.includes(`image/`)) {
+    return {
+      mimeType,
+      url,
+    }
+  }
```

Comment:
If mimetype is not image we can still use imageCDN, you don't need placeholderUrl or width & height.